### PR TITLE
lakectl: Add commit params to merge command

### LINKED ISSUE #6888: Add commit params to lakectl merge
Part of #6883

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #6888

### Background

Add commit parameters to the lakectl merge command so that user can modify the commit message and add metadata
      
### New Feature

Add missing functionality to lakectl

### Testing Details

Added system test

### Breaking Change?

No

### CODE REVIEW

File: cmd/lakectl/cmd/root.go [hunk 1]
```diff
@@ -208,6 +213,52 @@ func getSyncArgs(args []string, requireRemote bool, considerGitRoot bool) (remot
...
+	kvList, err := cmd.Flags().GetStringSlice(name)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]string)
+	for _, pair := range kvList {
+		key, value, found := strings.Cut(pair, "=")
+		if !found {
+			return nil, errInvalidKeyValueFormat
```
Comment (itaiad200) [inline cmd/lakectl/cmd/root.go:255]: I suggest being more informative:
```suggestion
			return nil, err.FormatErr("missing '=' in '%s': %v", pair, errInvalidKeyValueFormat)
```

File: cmd/lakectl/cmd/merge.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ var mergeCmd = &cobra.Command{
 		return validRepositoryToComplete(cmd.Context(), toComplete)
 	},
 	Run: func(cmd *cobra.Command, args []string) {
+		message, kvPairs := getCommitFlags(cmd)
```
Comment (itaiad200) [inline cmd/lakectl/cmd/merge.go:37]: Does `allow-empty-message` fit here? What's the default behavior that we now change?

File: cmd/lakectl/cmd/root.go [hunk 3]
```diff
@@ -208,6 +213,52 @@ func getSyncArgs(args []string, requireRemote bool, considerGitRoot bool) (remot
...
+	return message, kvPairs
+}
+
+func getKV(cmd *cobra.Command, name string) (map[string]string, error) { //nolint:unparam
+	kvList, err := cmd.Flags().GetStringSlice(name)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]string)
```
Comment (itaiad200) [inline cmd/lakectl/cmd/root.go:251]: Is nil different than an empty map? I want to make sure we're not going to show an empty map (`{}`) everywhere in the UI, CLI, etc.

File: cmd/lakectl/cmd/root.go [hunk 4]
```diff
@@ -208,6 +213,52 @@ func getSyncArgs(args []string, requireRemote bool, considerGitRoot bool) (remot
...
+func withMessageFlags(cmd *cobra.Command, defaultMsg string, allowEmpty bool) {
+	cmd.Flags().StringP(commitMsgFlagName, "m", defaultMsg, "commit message")
+	cmd.Flags().Bool(allowEmptyMsgFlagName, allowEmpty, "allow an empty commit message")
+}
+
+func withMetadataFlag(cmd *cobra.Command) {
+	cmd.Flags().StringSlice(metaFlagName, []string{}, "key value pair in the form of key=value")
+}
+
+func withCommitFlags(cmd *cobra.Command, defaultMsg string, allowEmptyMessage bool) {
```
Comment (itaiad200) [inline cmd/lakectl/cmd/root.go:225]: `allowEmptyMessage` is always false and also not sure we want to allow empty messages for merges at all

File: cmd/lakectl/cmd/import.go [hunk 5]
```diff
@@ -205,7 +201,6 @@ func init() {
 	importCmd.Flags().Bool("merge", false, "merge imported branch into target branch")
 	_ = importCmd.Flags().MarkDeprecated("merge", "import is done directly into target branch")
 	importCmd.Flags().Bool("no-progress", false, "switch off the progress output")
-	importCmd.Flags().StringP("message", "m", "Import objects", "commit message")
-	importCmd.Flags().StringSlice("meta", []string{}, "key value pair in the form of key=value")
+	withCommitFlags(importCmd, "Import objects", false)
```
Comment (itaiad200) [inline cmd/lakectl/cmd/import.go:204]: Now we allow empty messages in import also? I know it's a side effect of this PR but I'm not sure we want to add it to import and merge

File: esti/lakectl_test.go [hunk 6]
```diff
@@ -201,6 +201,51 @@ func TestLakectlBranchAndTagValidation(t *testing.T) {
...
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" commit lakefs://"+repoName+"/"+mainBranch+" -m \""+commitMessage+"\"", false, "lakectl_commit", vars)
+	featureBranch := "feature"
+	branchVars := map[string]string{
+		"REPO":          repoName,
+		"STORAGE":       storage,
+		"SOURCE_BRANCH": mainBranch,
+		"DEST_BRANCH":   featureBranch,
+	}
+
+	t.Run("merge with commit message and meta", func(t *testing.T) {
```
Comment (itaiad200) [inline esti/lakectl_test.go:230]: Can you add a test without commit message and meta? That's the only one that exists today and we need to double check we don't change the current behavior.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Thanks! Requesting changes since I want to verify we don't make any breaking changes to the current commit/import/merge/local-commit along the way. Notice that the refactor created some added functionality in this PR (like import empty message) and if we want to keep that, we need to add tests that current behavior is unchanged.

File: esti/lakectl_test.go (hunk 6 above)
Comment (N-o-Z) [reply esti/lakectl_test.go:230]: All the existing tests already run without commit and metadata. See (`TestLakectlMergeAndStrategies`)

File: cmd/lakectl/cmd/import.go (hunk 5 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/import.go:204]: Merge already handles empty messages with a default: "Merge '%s' into '%s'" message.
I will add the same behavior for import as well

File: cmd/lakectl/cmd/root.go (hunk 4 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:225]: There's a default merge message which we use when the message is empty (as was the current situation for merge in lakectl and UI)

File: cmd/lakectl/cmd/root.go (hunk 3 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:251]: The code did not change - just moved to a new file. The behavior remains the same as in commit today

File: cmd/lakectl/cmd/merge.go (hunk 2 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/merge.go:37]: Fixed to fallback to previous default behavior

File: cmd/lakectl/cmd/root.go (hunk 1 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:255]: The error is informative enough:
`invalid key/value pair - should be separated by "="`

File: cmd/lakectl/cmd/merge.go [hunk 7]
```diff
@@ -79,6 +79,6 @@ var mergeCmd = &cobra.Command{
 //nolint:gochecknoinits
 func init() {
 	mergeCmd.Flags().String("strategy", "", "In case of a merge conflict, this option will force the merge process to automatically favor changes from the dest branch (\"dest-wins\") or from the source branch(\"source-wins\"). In case no selection is made, the merge process will fail in case of a conflict")
-	withCommitFlags(mergeCmd, "Merge commit", false) // Added default msg to allow backwards compatibility
+	withCommitFlags(mergeCmd, true) // Added default msg to allow backwards compatibility
```
Comment (itaiad200) [inline cmd/lakectl/cmd/merge.go:82]: Is the comment still relevant?

**Review by itaiad200 — APPROVED** [review]

File: cmd/lakectl/cmd/merge.go (hunk 7 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/merge.go:82]: Good catch! 💪🏽

**Pull request merged**